File: src/main/kotlin/no/nav/aap/api/rest/sts/STSClientConfig.java
package no.nav.aap.api.rest.sts;

import no.nav.aap.api.rest.AbstractRestConfig;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;

import static no.nav.aap.api.config.Constants.STS;

@Configuration
public class STSClientConfig {

    @Bean
    @Qualifier(STS)
    public WebClient webClientSTS(WebClient.Builder builder, STSConfig cfg) {
        return builder
                .baseUrl(cfg.getBaseUri().toString())
                .filter(AbstractRestConfig.Companion.correlatingFilterFunction())
                .defaultHeaders(h -> h.setBasicAuth(cfg.getUsername(), cfg.getPassword()))
                .build();
    }
}

File: src/main/kotlin/no/nav/aap/api/rest/sts/STSClientConfig.kt
package no.nav.aap.api.rest.sts

import no.nav.aap.api.config.Constants.STS
import no.nav.aap.api.rest.AbstractRestConfig.Companion.correlatingFilterFunction
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpHeaders
import org.springframework.web.reactive.function.client.WebClient

@Configuration
internal class STSClientConfig {
    @Bean
    @Qualifier(STS)
    fun webClientSTS(builder: WebClient.Builder, cfg: STSConfig): WebClient {
        return builder
            .baseUrl(cfg.baseUri.toString())
            .filter(correlatingFilterFunction())
            .defaultHeaders { h: HttpHeaders -> h.setBasicAuth(cfg.username, cfg.password) }
            .build()
    }
}